src/convert.js

```diff
@@ -0,0 +1,222 @@
+/*!
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Buffer = require('safe-buffer').Buffer;
+const is = require('is');
+
+const validate = require('./validate')();
+
+/**
+ * Converts an ISO 8601 or Protobuf JS 'timestampValue' into Protobuf JS.
+ *
+ * @private
+ * @param {*=} timestampValue - The value to convert.
+ * @return {{nanos,seconds}|undefined} The value as expected by Protobuf JS or
+ * undefined if no input was provided.
+ */
+function convertTimestamp(timestampValue) {
+  let timestampProto = undefined;
+
+  if (is.string(timestampValue)) {
+    let date = new Date(timestampValue);
+    let seconds = Math.floor(date.getTime() / 1000);
+    let nanos = null;
+
+    let nanoString = timestampValue.substring(20, timestampValue.length - 1);
+
+    if (nanoString.length === 3) {
+      nanoString = `${nanoString}000000`;
+    } else if (nanoString.length === 6) {
+      nanoString = `${nanoString}000`;
+    }
+
+    if (nanoString.length === 9) {
+      nanos = parseInt(nanoString);
+    }
+
+    if (isNaN(seconds) || isNaN(nanos)) {
+      // This error should only ever be thrown if the end-user specifies an
+      // invalid 'lastUpdateTime'.
```

Comment:
This comment is meant to explain why I use "lastUpdateTime" and not "timestampValue" in the error message. I took this opportunity to expand on the comment since it obviously didn't quite convey this meaning :)

Reply:
Oh, I see!  This seems a bit fragile...  I'd expect to either pass in the parameter name as a string to use in the error message or to throw a generic error and have the calling code try/catch it and throw a more friendly error.  I'm fine with this as-is, but I'd be tempted to add "MINOR HACK: " or similar to the comment to make it clearer that we're taking a shortcut here...